homeassistant/components/thethingsnetwork/entity.py

```diff
@@ -0,0 +1,83 @@
+"""Support for The Things Network entities."""
+
+import logging
+
+from ttn_client import TTNSensorValue
+
+from homeassistant.core import callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import CONF_APP_ID, DOMAIN
+from .coordinator import TTNCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TTNEntity(CoordinatorEntity[TTNCoordinator]):
+    """Representation of a The Things Network Data Storage sensor."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: "TTNCoordinator",
+        ttn_value: TTNSensorValue,
+    ) -> None:
+        """Initialize a The Things Network Data Storage sensor."""
+
+        self._ttn_value = ttn_value
+
+        self._attr_unique_id = f"{self.device_id}_{self.field_id}"
+        self._attr_name = self.field_id
+
+        assert coordinator.config_entry
+        self._attr_device_info = DeviceInfo(
+            identifiers={
+                # Serial numbers are unique identifiers within a specific domain
+                (  # type: ignore [arg-type]
+                    DOMAIN,
+                    str(coordinator.config_entry.data[CONF_APP_ID]),
+                    self.device_id,
```

Comment:
What's an example of device id? Is it unique overall or per app?

Reply:
yes, each device has its own id such as `my_tracker`. Each TTN application has 0..n devices and each device 1..n entities.

Reply:
Maybe it makes sense to add app id as a parameter (so you don't have to access the config entry in every class again) and make the device identifiers {(DOMAIN, f"{app_id}_{device_id}")}

Reply:
ok, I like that as it will make the deviceInfo identifies type match. reworking code...

Reply:
We have had problems before because it didn't match so we try to avoid it when possible